scripts/functions/ae.py

```diff
@@ -0,0 +1,330 @@
+    # Initialize hyper parameters
+    #
+    # learning rate: 
+    # batch size: Total number of training examples present in a single batch
+    #             Iterations is the number of batches needed to complete one epoch
+    # epochs: One Epoch is when an ENTIRE dataset is passed forward and backward through the neural network only ONCE
+    # kappa: warmup
+    # original dim: dimensions of the raw data
+    # latent dim: dimensiosn of the latent space (fixed by the user)
+    #   Note: intrinsic latent space dimension unknown
+    # epsilon std: 
+    # beta: Threshold value for ReLU?
+    # --------------------------------------------------------------------------------------------------------------------
+
+    original_dim = rnaseq.shape[1]
+    beta = K.variable(0)
+
+    stat_file =  os.path.join(base_dir, "stats", analysis_name, "ae_2layer_{}latent_stats.tsv".format(latent_dim))
+    hist_plot_file =os.path.join(base_dir, "stats", analysis_name, "ae_2layer_{}latent_hist.png".format(latent_dim))
+
+    encoded_file =os.path.join(base_dir, "encoded", analysis_name, 
+                               "ae_train_input_2layer_{}latent_encoded.txt".format(latent_dim))
+
+    model_encoder_file =os.path.join(base_dir, "models", analysis_name, 
+                                     "ae_2layer_{}latent_encoder_model.h5".format(latent_dim))
+    weights_encoder_file =os.path.join(base_dir, "models", analysis_name, 
+                                       "ae_2layer_{}latent_encoder_weights.h5".format(latent_dim))
+    model_decoder_file =os.path.join(base_dir, "models", analysis_name, 
+                                     "ae_2layer_{}latent_decoder_model.h5".format(latent_dim))
+    weights_decoder_file =os.path.join(base_dir, "models", analysis_name,
+                                       "ae_2layer_{}latent_decoder_weights.h5".format(latent_dim))
+
+
+    # --------------------------------------------------------------------------------------------------------------------
+    # Functions
+    #
+    # Based on publication by Greg et. al. 
+    # https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5728678/
+    # https://github.com/greenelab/tybalt/blob/master/scripts/vae_pancancer.py
+    # --------------------------------------------------------------------------------------------------------------------
+
+    # Function for reparameterization trick to make model differentiable
+    def sampling(args):
+
+        # Function with args required for Keras Lambda function
+        z_mean, z_log_var = args
+
+        # Draw epsilon of the same shape from a standard normal distribution
+        epsilon = K.random_normal(shape=tf.shape(z_mean), mean=0.,
+                                  stddev=epsilon_std)
+
+        # The latent vector is non-deterministic and differentiable
+        # in respect to z_mean and z_log_var
+        z = z_mean + K.exp(z_log_var / 2) * epsilon
+        return z
+
+
+    class CustomVariationalLayer(Layer):
```

Comment:
I think this was some legacy code that I scrapped from you a while back and since then you have cleaned up your code to do exactly as you said because its much cleaner that way.  I had meant to change this but didn't get time so its good you pointed this out